app/src/main/java/org/p2p/wallet/utils/StdExtensions.kt

```diff
@@ -67,3 +72,45 @@ fun ChaCha20Poly1305.processBytesKt(
 fun BigDecimal?.compareTo(other: BigDecimal?): Int = this.orZero().compareTo(other.orZero())
 
 fun JsonArray.getOrNull(index: Int): JsonElement? = if (index >= size()) null else get(index)
+
+/**
+ * Retries [block] if it throws an exception of type [exceptionTypes] up to [maxAttempts] times.
+ * When [maxAttempts] is reached or the exception is not of type [exceptionTypes], the exception is rethrown.
+ *
+ * @param exceptionTypes the types of exceptions to retry on
+ * (default: [SocketTimeoutException], [InterruptedIOException])
+ * @param maxAttempts the maximum number of attempts to retry (default: 3)
+ * @param delayMillis the delay between attempts (default: 1 second)
+ * @param block the block to execute
+ * @return the result of [block]
+ */
+suspend inline fun <T> retryOnException(
```

Comment:
can you put it near org.p2p.wallet.utils.CoroutinesExtensionsKt#retryRequest ?

Reply:
Rocket science function x D